src/large_pages/node_large_page.cc

```diff
@@ -282,6 +282,25 @@ bool IsSuperPagesEnabled() {
 }
 #endif
 
+class MmapPointer {
+ public:
+  void
+  DoMmap(void* start, size_t size_, int prot, int flags, int fd, int offset) {
```

Comment:
I feel like it makes sense to a) provide default arguments here and b) also add a constructor that does this for easier use?

Reply:
Agreed on both points

Reply:
a) The only parameter that is always 0 is the `offset`. Even the `fd` is not always -1 – see the `__APPLE__` case. It surprised me BTW that apple uses some predefined constant as the file descriptor 🤨. Seems like an abuse of the parameter.

b) The problem with doing mmap in the constructor is that it requires the declaration to be placed between `goto fail` and `fail:`, which won't compile.

Reply:
Move and copy constructors would address that, allowing the declaration as it is currently then `mmap = MmapPointer(...)` inline. I agree with @addaleax that it's a bit nicer API wise and is consistent with how we handle it elsewhere.

Reply:
For (a), setting an appropriate constexpr for `fd` depending on each platform then defaulting both `fd` and `offset` would seem to be appropriate?

Reply:
We've moved in a different direction given @bnoordhuis' comments. We now `mmap(2)` outside the class.